### PR TITLE
Fix leak memory and upgrade csi-lib-util dependency

### PR DESCRIPTION
**Pull request by Jiawei0227**
**What type of PR is this?**
> /kind bug

**What this PR does / why we need it**:
This PR does the following two things:

1. It fixes a bug where memory get leaked because every time there is a new http request, it will create a csiMetricsManager. That is unnecessary and can lead to increasing memory allocated.
2. This PR upgrades the csi-lib-utils to v0.9.0

**Which issue(s) this PR fixes**:

Fixes # https://github.com/kubernetes-csi/livenessprobe/issues/91

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Fix a memory leak issue in livenessprobe. Upgrade csi-lib-utils dependency to v0.9.0
```

### CODE REVIEW

File: cmd/livenessprobe/main.go [hunk 1]
```diff
@@ -42,14 +42,15 @@ var (
...
+	driverName     string
+	metricsManager *metrics.CSIMetricsManager
 }
 
 func (h *healthProbe) checkProbe(w http.ResponseWriter, req *http.Request) {
 	ctx, cancel := context.WithTimeout(req.Context(), *probeTimeout)
 	defer cancel()
 
-	conn, err := acquireConnection(ctx, metrics.NewCSIMetricsManager(""))
+	conn, err := acquireConnection(ctx, *h.metricsManager)
```
Comment (Jiawei0227) [inline cmd/livenessprobe/main.go:53]: This is where we fix the memory leak issue.

File: go.mod [hunk 2]
```diff
@@ -5,11 +5,8 @@ go 1.15
 require (
 	github.com/container-storage-interface/spec v1.3.0
 	github.com/golang/mock v1.4.3
-	github.com/golang/protobuf v1.4.2 // indirect
-	github.com/kubernetes-csi/csi-lib-utils v0.7.0
+	github.com/kubernetes-csi/csi-lib-utils v0.9.0
```
Comment (msau42) [inline go.mod:8]: can you also check that all uses of klog in this repo are switched to klog v2?

**Comment by msau42** [comment]
/approve
will let @verult review

File: go.mod (hunk 2 above)
Comment (Jiawei0227) [reply go.mod:8]: It seems not using klog v1 anywhere in the code.

File: cmd/livenessprobe/main.go [hunk 3]
```diff
@@ -42,14 +42,15 @@ var (
 )
 
 type healthProbe struct {
-	driverName string
+	driverName     string
+	metricsManager *metrics.CSIMetricsManager
```
Comment (verult) [inline cmd/livenessprobe/main.go:46]: Does this need to be a pointer? The returned object is already a struct pointer but with an interface type

Comment (Jiawei0227) [reply cmd/livenessprobe/main.go]: Good point! Fixed!

**Pull request merged**